src/screens/CreateContactScreen.tsx

```diff
@@ -50,37 +59,58 @@ export class CreateContactScreen extends React.PureComponent<Props, State> {
     return !!this.state.address && !!this.state.name;
   }
 
-  setName = (name: string) => this.setState({ name });
+  setName = (value: string) => this.setState({ name: { value, error: '' } });
 
-  setAddress = (address: string) => this.setState({ address, error: '' });
+  setAddress = (value: string) => this.setState({ address: { value, error: '' } });
 
   onBarCodeScan = (address: string) => {
     this.setAddress(address.split('?')[0].replace('bitcoin:', ''));
   };
 
   createContact = () => {
-    try {
-      this.validateAddress();
-      if (this.state.error) return;
-      this.props.createContact({
-        id: uuidv4(),
-        name: this.state.name.trim(),
-        address: this.state.address.trim(),
-      });
-      this.showSuccessImportMessageScreen();
-      this.setState({
-        name: '',
-        address: '',
-      });
-    } catch (_) {
+    const { name, address } = this.state;
+    const nameError = this.validateName();
+    const addressError = this.validateAddress();
+
+    if (addressError || nameError) {
       this.setState({
-        error: i18n.send.details.address_field_is_not_valid,
+        name: { value: name.value, error: nameError },
+        address: { value: address.value, error: addressError },
       });
+      return;
     }
+    this.props.createContact({
+      id: uuidv4(),
+      name: name.value.trim(),
+      address: address.value.trim(),
+    });
+    this.showSuccessImportMessageScreen();
+    this.setState({
```

Comment:
is there a warning that you shouldn't change state after the component did unmount?

Reply:
No, there's no warning, but I will change the order of set state, to make it before the navigate

Reply:
sure, good point :+1: